### ISSUE TITLE
dask supports distributed training on databricks clusters ? 

### ISSUE THREAD

**Issue by usact** [dask/distributed#3533]
I am trying to run an example of dask at https://examples.dask.org/machine-learning.html on databricks.

My python 3  code: 

       from dask.distributed import Client, progress 
       client = Client(processes=False, threads_per_worker=8,
                n_workers=2, memory_limit='6GB')
        client

I am not sure why it always has this kind of complaining ?

-----------

 distributed.worker - WARNING - Memory use is high but worker has no data to store to disk.  Perhaps some other process is leaking memory?  Process memory: 5.31 GB -- Worker memory limit: 4.00 GB

--------------

      from sklearn.datasets import make_classification
      from sklearn.svm import SVC
      from sklearn.model_selection import GridSearchCV
      import pandas as pd

       X, y = make_classification(n_samples=1000000, random_state=0)

       param_grid = {"C": [0.001, 0.01, 0.1, 0.5, 1.0, 2.0, 5.0, 10.0],
              "kernel": ['rbf', 'poly', 'sigmoid'],
              "shrinking": [True, False]}

       grid_search = GridSearchCV(SVC(gamma='auto', random_state=0, probability=True),
                           param_grid=param_grid,
                           return_train_score=False,
                           iid=True,
                           cv=3,
                           n_jobs=-1)

       import joblib

       with joblib.parallel_backend('dask'):
            grid_search.fit(X, y)

When I checked the cpu usage from ganglia on databricks cluster that has 1 driver and 1 - 64 workers, only the driver is busy and all worker machines are idle. 

The cluster enables auto-scaling.

Why only the driver is busy ?

Also, I have tried:

      from dask_jobqueue import PBSCluster, HTCondorCluster, JobQueueCluster, LSFCluster, MoabCluster, OARCluster, SGECluster, SLURMCluster

     cluster = SLURMCluster(cores=16,
                     memory="1GB",
                     disk='1GB',
                     project='test_project',
                     queue='premium',
                     interface='eth0')# 'ib0',
                     #job_cls=SLURMJob)
                     #walltime='02:00:00')

     cluster.scale(100)  # Start 100 workers in 100 jobs that match the description above

     from dask.distributed import Client
     client = Client(cluster) 

But, it does not work. 

thanks

**Comment by mrocklin** [comment]
Databricks is running Spark, not Dask.  In order to get distributed computation with Dask-ML you need to be running Dask.

**Comment by usact** [comment]
I have installed dask on databricks cluster.

**Comment by usact** [comment]
also, dask_jobqueue.

**Comment by mrocklin** [comment]
Unfortunately that isn't enough.  Databricks does not provide multi node
Dask.

**Comment by usact** [comment]
Thanks for your answer. So, it means that it is not possible to run parallel dask jobs on multiple nodes of databricks ? And, dask can only be run on one single machine by multi-threading ?

**Comment by TomAugspurger** [comment]
I’d expect it to work as long as you configure things correctly. I wouldn’t expect it to work out of the box though.

**Comment by jakirkham** [comment]
Happy to close @usact? 🙂

**Comment by TomAugspurger** [comment]
Looks like it.
